src/openrct2/ride/TrackDesignRepository.cpp

```diff
@@ -78,10 +78,15 @@ class TrackDesignFileIndex final : public FileIndex<TrackRepositoryItem>
         auto td6 = track_design_open(path.c_str());
         if (td6 != nullptr)
         {
+            auto* rawObject = object_repository_load_object(&td6->vehicle_object);
```

Comment:
This is causing a race condition when building the track design list (looks to be issues with the image list). I don't think you should be loading and unloading objects in threads. You will either have to have this protected by a mutex or fix object loading / unloading to be thread safe.

Reply:
Don't think I can pull off the latter, so I'd want to do the former. How do I introduce one? Is there any other place in OpenRCT2 that uses a mutex?

Reply:
Yes, just search for `std::mutex`. I think the object downloader and server list use them.

Reply:
I have now added one. I hope I did it correctly.